extensions/origins/source/protocols/pubsub/Topic.ts

```diff
@@ -0,0 +1,23 @@
+import { Connector } from '@toa.io/core'
+import type { Origin } from './Origin'
+import type { PublishOptions, PubSub } from '@google-cloud/pubsub'
+
+export class Topic extends Connector {
+  private readonly publisher
+
+  public constructor (pubsub: PubSub, origin: Origin) {
+    super()
+
+    const name = `projects/${origin.project}/topics/${origin.topic}`
+
+    this.publisher = pubsub.topic(name, OPTIONS)
+  }
+
+  public async publish (payload: unknown): Promise<string> {
+    const data = Buffer.from(JSON.stringify(payload))
+
+    return this.publisher.publishMessage({ data })
+  }
+}
+
+const OPTIONS: PublishOptions = { batching: { maxMessages: 100, maxMilliseconds: 1000 } }
```

Comment:
It's a little bit unusual for a framework to not let adjust these options. You basically rate limited pubsub to 1 call per second and that may not fit many use cases

Reply:
it does not limit the rate, it limits the batch accumulation.

Reply:
I maybe misunderstand how batching works in GCP but from what I understood it means it will either send when we have 100 messages or after 1 second after the first message. That for scenarios with less than 100 messages per second means that it will not do more that 1 call per second

Reply:
In case when there is less than 100 messages per second, yes.
I believe this is inevitable for any batching algorithm.

Reply:
My question was why we are hardcoding this in the framework without providing any mean of customising those settings?

Reply:
Technically, for the same reason as below: currently Origin is configured with a URL. I could have added something like query parameters to it (similar to MongoDB), but I don't want to propagate the issue further.

Once Origins are refactored from URLs to objects, there will be a straightforward solution to add any settings.